When selecting Live Text in Safari on iOS, the selection handles move away from the selection when zooming

rr-codes opened the pull request:
#### 69d07a6b99c8a8d16460bcbb220cb68c63758479
<pre>
When selecting Live Text in Safari on iOS, the selection handles move away from the selection when zooming
<a href="https://bugs.webkit.org/show_bug.cgi?id=259964">https://bugs.webkit.org/show_bug.cgi?id=259964</a>
rdar://111793617

Reviewed by Wenson Hsieh.

UIKit now scales the selection handles when zooming and updates their position accordingly.
Since WebKit is also scaling them ourselves, this results in a position that&apos;s offset from the
selection proportional to the zoom amount.

Fix by no longer scaling the selection handles in WebKit.

* LayoutTests/resources/ui-helper.js:
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription.return.new.Promise.):
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription.return.new.Promise):
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription):
* Source/WebKit/UIProcess/ios/WKTextSelectionRect.mm:
(-[WKTextSelectionRect _customHandleInfo]):
* Tools/TestRunnerShared/UIScriptContext/Bindings/UIScriptController.idl:
* Tools/TestRunnerShared/UIScriptContext/UIScriptController.h:
(WTR::UIScriptController::selectionEndGrabberViewShapePathDescription const):
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::selectionEndGrabberViewShapePathDescription const):

Canonical link: <a href="https://commits.webkit.org/266753@main">https://commits.webkit.org/266753@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2b140beff694c1a6247b1cb19166b1591b6dd969

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14691 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15001 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15350 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16439 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14829 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17519 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15084 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14872 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15375 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/12474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17174 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/12658 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13252 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/13733 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13419 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/16662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13964 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/11809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13265 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3533 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17600 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13810 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
```diff
@@ -882,6 +922,22 @@ static void clipSelectionViewRectToContentView(CGRect& rect, UIView *contentView
+    UIView *handleView = nil;
+
+#if HAVE(UI_TEXT_SELECTION_DISPLAY_INTERACTION)
+    UNUSED_PARAM(contentView);
```
Nit - I would remove this and just replace the `contentView` below with `platformContentView()`.

rr-codes committed:
When selecting Live Text in Safari on iOS, the selection handles move away from the selection when zooming
https://bugs.webkit.org/show_bug.cgi?id=259964
rdar://111793617

Reviewed by Wenson Hsieh.

UIKit now scales the selection handles when zooming and updates their position accordingly.
Since WebKit is also scaling them ourselves, this results in a position that's offset from the
selection proportional to the zoom amount.

Fix by no longer scaling the selection handles in WebKit.

* LayoutTests/resources/ui-helper.js:
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription.return.new.Promise.):
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription.return.new.Promise):
(window.UIHelper.getSelectionEndGrabberViewShapePathDescription):
* Source/WebKit/UIProcess/ios/WKTextSelectionRect.mm:
(-[WKTextSelectionRect _customHandleInfo]):
* Tools/TestRunnerShared/UIScriptContext/Bindings/UIScriptController.idl:
* Tools/TestRunnerShared/UIScriptContext/UIScriptController.h:
(WTR::UIScriptController::selectionEndGrabberViewShapePathDescription const):
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::selectionEndGrabberViewShapePathDescription const):

Canonical link: https://commits.webkit.org/266753@main

pxlcoder commented on Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
```diff
@@ -106,6 +106,46 @@ static BOOL returnNo()
+        }
+    };
+
+    auto attributes = adoptNS([[NSMutableDictionary alloc] init]);
```
This really should have been an array of key-value pairs, since it's very possible to have the same path element type show up multiple times in a given path.
